### PR TITLE
React hotloading

### PR DESCRIPTION
**Pull request by VasilyShelkov**
Good news, finally got this working ! this is for #15

@Winwardo, ready for merging :) you're welcome to test, although until you do any front-end dev you won't notice a difference.

I've tested by 
for prod:

```
npm build
npm start
```

for testing: `npm run test-quick` and all passed

for dev: `npm run dev` which restarts the server when changes are made to server code and hot loads the changes to the browser when changes are made to the client :D so happy

### CODE REVIEW

File: src/client/App.js [hunk 1]
```diff
@@ -5,7 +5,7 @@ const App = () => {
 		<div>
 			<h2 className="ui center aligned icon header">
 				<i className="circular users icon"></i>
-  			Socto
+  			Socto I win at life
```
Comment (Winwardo) [inline src/client/App.js]: :+1:

File: src/server/server.js [hunk 2]
```diff
@@ -6,8 +8,23 @@ import { searchAndSave } from './twitterSearch';
 const app = express();
 const port = process.env.PORT || 3000;
 
+//In development hotload React using webpack-hot-middleware
+if (!(process.env.NODE_ENV === 'production')){
```
Comment (Winwardo) [inline src/server/server.js]: Indentation here is screwed, did you npm run lint?

File: webpack.config.js [hunk 3]
```diff
@@ -3,7 +3,7 @@ var fs = require('fs');
 const webpack = require('webpack');
 const extend = require('extend');
 
-const production = process.env.NODE_ENV === 'production' ? false : true;
+const production = process.env.NODE_ENV === 'production' ? true : false;
```
Comment (Winwardo) [inline webpack.config.js]: Remove the `? true : false`.
Should be just `const production = process.env.NODE_ENV === 'production';`

**Comment by Winwardo** [comment]
> npm run dev

```
C:\Users\Topher\Documents\GitHub\solid-octo-disco\build\server.js:75
          app.use(__webpack_require__(!(function webpackMissingModule() { var e = new Error("Cannot find module \"webpack-dev-middleware\""); e.code = 'MODULE_NOT_FOUND'; throw e; }()))(compiler, {
                                                                                                                                                                           ^

Error: Cannot find module "webpack-dev-middleware"
    at webpackMissingModule (C:\Users\Topher\Documents\GitHub\solid-octo-disco\build\server.js:75:76)
    at Object.<anonymous> (C:\Users\Topher\Documents\GitHub\solid-octo-disco\build\server.js:75:175)
    at __webpack_require__ (C:\Users\Topher\Documents\GitHub\solid-octo-disco\build\server.js:20:30)
    at C:\Users\Topher\Documents\GitHub\solid-octo-disco\build\server.js:40:18
    at Object.<anonymous> (C:\Users\Topher\Documents\GitHub\solid-octo-disco\build\server.js:43:10)
    at Module._compile (module.js:434:26)
    at Object.Module._extensions..js (module.js:452:10)
    at Module.load (module.js:355:32)
    at Function.Module._load (module.js:310:12)
    at Function.Module.runMain (module.js:475:10)
    at startup (node.js:117:18)
    at node.js:951:3
```

**Comment by Winwardo** [comment]
On second `npm run dev` it worked and is currently hotloading correctly.
Strange.

**Comment by VasilyShelkov** [comment]
you probably need to do npm install...or remove the node modules and redo it.

@Winwardo fixed some lint errors...

**Pull request merged**